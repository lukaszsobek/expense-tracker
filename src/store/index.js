import {
    applyMiddleware,
    createStore,
    combineReducers
} from "redux"

import thunk from "redux-thunk"

import {
    expensesReducer,
    filtersReducer,
    authReducer
} from "../reducers"

export default () => {
    const store = createStore(
        combineReducers({
            expenses: expensesReducer,
            filters: filtersReducer,
            auth: authReducer
        }),
        applyMiddleware(thunk)
    )
    return store
}