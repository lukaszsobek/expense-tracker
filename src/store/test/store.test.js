import store from "../"

describe("Store",() => {

    it("has correct initial state",() => {
        expect(store().getState()).toMatchSnapshot()
    })
})