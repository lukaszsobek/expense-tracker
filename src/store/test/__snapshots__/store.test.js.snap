// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store has correct initial state 1`] = `
Object {
  "expenses": Array [],
  "filters": Object {
    "endDate": "1970-01-31T22:59:59.999Z",
    "sortBy": "date",
    "startDate": "1969-12-31T23:00:00.000Z",
    "text": "",
  },
}
`;
